### DISCUSSION TITLE
Return receipt - Item scanning extension

### DISCUSSION THREAD

**Discussion by ahgamal-ms** [microsoft/Dynamics365Commerce.Solutions#172] [Q&A] [answered]
We have a customized bar code that include item batch number with item [ itemid-batchNo ] (0001-9988). we have implemented GetScanResultClientRequestHandlerExt to extract the batch no and save that as extension property on car line.

with receipt return when receipt has same item twice we need to be able to scan the returned product and then match the return line with the batch no from extension property.So the expectations is to select the line for return instead of prompt the user to reselect it.
 
 
![image](https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/74003164/e99608f4-b5b9-4a61-97b5-33c9e67f4759)

 
We have tried to look into MS code and can see match by product record id and in above case we have multiple records returned. 
 
So the question is it possible to extend this? and  to customized the matching here to be with record id + Batch no form extension prop.

**Comment by madyke** [comment, marked as answer]
Hi @ahgamal-ms - Luke and I looked into this request this morning, but unfortunately this is not something that is possible with the current extension points. If the customer wants this functionality they can submit an extensibility request:
[Extensibility requests - Finance & Operations | Dynamics 365 | Microsoft Learn](https://learn.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/extensibility/extensibility-requests)
